Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py

```diff
@@ -0,0 +1,193 @@
+from typing import Dict, Any
+
+from imapclient import IMAPClient
+from imap_tools import OR
+import mailparser
+import demistomock as demisto
+from CommonServerPython import *  # noqa: E402 lgtm [py/polluting-import]
+import dateparser
+from datetime import timezone
+
+# CONSTS
+INCLUDE_RAW_BODY = demisto.params().get('Include_raw_body', False)
+PERMITTED_FROM_ADDRESSES = demisto.params().get('permittedFromAdd', '')
+PERMITTED_FROM_DOMAINS = demisto.params().get('permittedFromDomain', '')
+DELETE_PROCESSED = demisto.params().get("delete_processed", False)
+
+
+class Email(object):
+    def __init__(self, email_object):
+        self.to = [mail_addresses for _, mail_addresses in email_object.to]
+        self.cc = [mail_addresses for _, mail_addresses in email_object.cc]
+        self.bcc = [mail_addresses for _, mail_addresses in email_object.bcc]
+        self.attachments = email_object.attachments
+        self.from_ = [mail_addresses for _, mail_addresses in email_object.from_][0]
+        self.format = email_object.message.get_content_type()
+        self.html = email_object.text_html[0] if email_object.text_html else ''
+        self.text = email_object.text_plain[0] if email_object.text_plain else ''
+        self.subject = email_object.subject
+        self.headers = email_object.headers
+        self.raw_body = email_object.body if INCLUDE_RAW_BODY else None
+        # According to the mailparser documentation the datetime object is in utc
+        self.date = email_object.date.replace(tzinfo=timezone.utc)
+        self.raw_json = self.generate_raw_json()
+
+    def parse_attachment(self):
+        file_names = []
+        for attachment in self.attachments:
+            payload = attachment.get('payload')
+
+            file_data = base64.b64decode(payload) if attachment.get('binary') else payload
+
+            # save the attachment
+            file_result = fileResult(attachment.get('filename'), file_data, attachment.get('mail_content_type'))
+
+            # check for error
+            if file_result['Type'] == entryTypes['error']:
+                demisto.error(file_result['Contents'])
+                raise Exception(file_result['Contents'])
```

Comment:
not sure we want to raise an exception here

Reply:
You think demisto.error is enough?

Reply:
yes